### PR TITLE
add always-true policy definition extension

### LINKED ISSUE #91: Add Default Policy: Always True
# Feature Request

## Description

As I understand, every asset offering needs a policy. As we just have some restricting policies, we need a default "always-true" policy, to avoid restrictions.

## Why Is the Feature Desired?

Customers do not want to publish assets always with a limiting constraint.

## Stakeholders

@SebastianOpriel

## Solution Proposal and Work Breakdown

- [x] Add always true policy.
- [x] Automatically add always-true policy on startup.
- [x] Add to open source images.
- [ ] Add to closed source images.
- [ ] E2E Test all images.

### PR DESCRIPTION
**Pull request by richardtreier**
related to #91

### CODE REVIEW

File: extensions/policy-always-true/src/test/java/de/sovity/edc/extension/policy/AlwaysTruePolicyExtensionTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import static java.util.Objects.requireNonNull;
+
+@ApiTest
+@ExtendWith(EdcExtension.class)
+class AlwaysTruePolicyExtensionTest {
+
+    @Test
+    void alwaysTruePolicyDef(PolicyEngine policyEngine, PolicyDefinitionService policyDefinitionService) {
+        // arrange
+        PolicyDefinition alwaysTrue = alwaysTruePolicy(policyDefinitionService);
```
Comment (efiege) [inline extensions/policy-always-true/src/test/java/de/sovity/edc/extension/policy/AlwaysTruePolicyExtensionTest.java:42]: var

File: extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public boolean exists() {
+        return policyDefinitionService.findById(POLICY_DEFINITION_ID) != null;
+    }
+
+    /**
+     * Creates policy definition &quot;always-true&quot;.
+     */
+    public void create() {
+        AtomicConstraint alwaysTrueConstraint = AtomicConstraint.Builder.newInstance()
```
Comment (efiege) [inline extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java:53]: var

File: extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        this.policyDefinitionService = policyDefinitionService;
+    }
+
+    /**
+     * Checks if policy definition &quot;always-true&quot; exists
+     *
+     * @return if exists
+     */
+    public boolean exists() {
+        return policyDefinitionService.findById(POLICY_DEFINITION_ID) != null;
```
Comment (efiege) [inline extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java:46]: Suggestion: AlwaysTruePolicyConstants.POLICY_DEFINITION_ID

So one could directly see that it is the AlwaysTruePolicy's definition Id

**Review by efiege — CHANGES_REQUESTED** [review]

File: extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java (hunk 3 above)
Comment (richardtreier) [reply extensions/policy-always-true/src/main/java/de/sovity/edc/extension/policy/services/AlwaysTruePolicyDefinitionService.java:46]: makes lines very long, fortunately in the IDE it's easy to comprehend where it's from

**Review by efiege — APPROVED** [review]

**Pull request merged**